Add introduced field to dataset coverage

jbothma opened the pull request:
In opensanctions/crawler-planning#76 we'd like to represent the date a dataset was first created in its metadata.

Temporal coverage tends to speak to the contents of the dataset.

Our datasets are normally point in time snapshots rather than time series covering the period since when the dataset was introduced. `coverage.end` is used well to indicate that the contents will not cover data later than that date. `coverage.start` would make sense to indicate e.g. the start of historical data in our dataset, e.g. if a source includes PEP position occupancy starting in 2004, or the age of the oldest sanction in the dataset.

I'd prefer not using `coverage.start` to represent the introduction of the dataset, because users might believe that we include historical data when we don't. Some datasets are also based on snapshots earlier than the date the data was added to opensanctions.

To avoid confusion with the date a particular release was created, I think `coverage.introduced` works well to describe what we're saying, rather than `coverage.created`.

pudo commented:
But to be clear: we are using `coverage.end` to document our observation of sources right now. So is your point to introduce a "philosophical coverage" (united states congress started in 1776) vs. "practical" coverage (e.g. the values we have in `coverage.end`)? My concern is that I am not sure what the value of documenting the "philosophical" values is, other than as entertaining trivia.

jbothma commented:
Not going for philosophical coverage, I don't think. I just think coverage.start as in the specs you referenced doesn't feel like it quite applies to (a) what we're publishing when we set it to when the dataset was added and (b) our use case.

The temporal coverage metadata in other specs tend to refer to the data in the dataset. Our use of coverage.end is consistent with that. 

I have two smallish issues with using coverage.start of when we added the dataset:

- as a user, if I see `coverage.start: 2023` for a parliamentary dataset that actually only includes current politicians, I might be surprised after the 2024 elections that the people who left office in 2024 aren't there anymore. The dataset says it covers that period.
- sometimes we add a dataset after the research was done. E.g. zvezoskop final coverage date is 2024-02-01 but we added it 2024-02-28. But imagine we're a year late to some really cool research completed 2023-02-20. The appropriate coverage start date is < = 2023-02-20 but what we are wanting to express here is that we added the dataset 2024-02-01.

I thought we were primarily interested in when we added the dataset, but do you really mean to express the coverage of the contents?
My issue with using the dataset added date for that is that we're not publishing time series data in the sense that we're not including all observations we ever saw since we added the dataset. Although there's an argument that first_seen means we can say "this dataset includes data that is current as of the release date, indicating when each item was first seen, which would be within the coverage start and end"
